File: tests/javascript/topologicalsort.test.js
const fs = require('fs');
const path = require('path');

const algoDir = path.join(__dirname, '../../topological-sort');
const possibleFiles = ['topological_sort.js', 'topological-sort.js', 'topologicalSort.js'];
let code = null;

for (const file of possibleFiles) {
  try {
    code = fs.readFileSync(path.join(algoDir, file), 'utf8');
    break;
  } catch (e) {}
}

if (!code) throw new Error('Could not find implementation file');

// Remove test code that runs automatically
const originalLog = console.log;
console.log = () => {};

const func = new Function(code + '; return topologicalSortDFS || topologicalSort;');
console.log = originalLog;
const topologicalSort = func();

describe('Topological Sort', () => {
  test('basic graph', () => {
    const graph = {5: [2, 0], 4: [0, 1], 2: [3], 3: [1], 0: [], 1: []};
    const result = topologicalSort(graph, 6);
    expect(Array.isArray(result)).toBe(true);
    expect(result.length).toBe(6);
  });
});

File: tests/java/TopologicalSortTest.java
import java.util.*;

public class TopologicalSortTest {
    public static void main(String[] args) {
        Map<Integer, List<Integer>> graph = new HashMap<>();
        graph.put(5, Arrays.asList(2, 0));
        graph.put(4, Arrays.asList(0, 1));
        graph.put(2, Arrays.asList(3));
        graph.put(3, Arrays.asList(1));
        graph.put(0, new ArrayList<>());
        graph.put(1, new ArrayList<>());
        
        List<Integer> result = TopologicalSort.topologicalSortDFS(graph, 6);
        assert result.size() == 6 : "Topological sort failed";
        System.out.println("TopologicalSortTest passed");
    }
}
